features/telemetry/protocols/netflow/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/netflow/adapter/netflow9/Netflow9Flow.java

```diff
@@ -275,6 +276,18 @@ public Integer getVlan() {
                 .orElse(null);
     }
 
+    @Override
+    public Optional<Timeout> getTimeout() {
+        final Optional<Long> active = getInt64(this.document, "FLOW_ACTIVE_TIMEOUT").map(t -> t * 1000L);
+        final Optional<Long> inactive = getInt64(this.document, "FLOW_INACTIVE_TIMEOUT").map(t -> t * 1000L);
+
+        if (active.isPresent() && inactive.isPresent()) {
+            return Optional.of(new Timeout(active.get(), inactive.get()));
+        } else {
+            return Optional.empty();
+        }
+    }
```

Comment:
Should this entire function move to `UpdatingFlow`?

Reply:
Then we have to extract two functions to return the different key names. I would like to keep it there to have the flexibility to have protocol specific logic.